go/saltpack/encrypt.go

```diff
@@ -149,17 +150,19 @@ func (es *encryptStream) init(sender BoxSecretKey, receivers []BoxPublicKey) err
 
 	eh.SenderSecretbox = secretbox.Seal([]byte{}, sender.GetPublicKey().ToKID(), (*[24]byte)(nonceForSenderKeySecretBox()), (*[32]byte)(&es.payloadKey))
 
-	for _, receiver := range receivers {
-		payloadKeyBox := ephemeralKey.Box(receiver, nonceForPayloadKeyBox(), es.payloadKey[:])
+	if !es.anonymous {
+		for _, receiver := range receivers {
+			payloadKeyBox := ephemeralKey.Box(receiver, nonceForPayloadKeyBox(), es.payloadKey[:])
 
-		keys := receiverKeys{PayloadKeyBox: payloadKeyBox}
+			keys := receiverKeys{PayloadKeyBox: payloadKeyBox}
 
-		// Don't specify the receivers if this public key wants to hide
-		if !receiver.HideIdentity() {
-			keys.ReceiverKID = receiver.ToKID()
-		}
+			// Don't specify the receivers if this public key wants to hide
+			if !receiver.HideIdentity() {
+				keys.ReceiverKID = receiver.ToKID()
```

Comment:
This would require setting creating new `BoxPublicKey`s for each receiver with a `HideIdentity() -> true` method. It's really an encryption option, not tied to the specific receivers.